### PR TITLE
implement ShoppingCartingcart and Ticket models, services and dao

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

File: src/main/java/project/cinema/Main.java [hunk 1]
```diff
@@ -53,5 +54,11 @@ public static void main(String[] args) throws AuthenticationException {
         authenticationService.register("bobos@gmail.com", "1123");
         User bob = authenticationService.login("bobos@gmail.com", "1123");
         System.out.println(bob);
+
+        ShoppingCartService shoppingCartService = (ShoppingCartService)
+                injector.getInstance(ShoppingCartService.class);
+        shoppingCartService.addSession(movieSession, bob);
+        shoppingCartService.addSession(movieSession, bob);
+        System.out.println("Bobs shopping card: " + shoppingCartService.getByUser(bob));
     }
```
Comment (AndriiBardin) [inline src/main/java/project/cinema/Main.java:63]: add shopping cart clear in main

Comment (AndriiBardin) [inline src/main/java/project/cinema/Main.java:67]: add shopping cart clear in main

**Review by AndriiBardin — CHANGES_REQUESTED** [review]

File: src/main/java/project/cinema/Main.java (hunk 1 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/Main.java:67]: added

**Review by AndriiBardin — APPROVED** [review]

File: src/main/java/project/cinema/Main.java [hunk 2]
```diff
@@ -53,5 +55,14 @@ public static void main(String[] args) throws AuthenticationException {
...
         System.out.println(bob);
+
+        ShoppingCartService shoppingCartService = (ShoppingCartService)
+                injector.getInstance(ShoppingCartService.class);
+        shoppingCartService.addSession(movieSession, bob);
+        shoppingCartService.addSession(movieSession, bob);
+        ShoppingCart bobShopCard = shoppingCartService.getByUser(bob);
+        System.out.println("Bobs shopping card: " + bobShopCard);
+        shoppingCartService.clear(bobShopCard);
+        System.out.println("clear: " + bobShopCard);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/Main.java]:
```suggestion
        System.out.println("clear: " + shoppingCartService.getByUser(bob));
```

File: src/main/java/project/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.update(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert shoppingCart: "
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't update shoppingCart: "
```

File: src/main/java/project/cinema/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, tickets, user);
+    }
+
+    @Override
+    public String toString() {
+        return "ShoppingCard{"
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/model/ShoppingCart.java]:
```suggestion
        return "ShoppingCart{"
```

File: src/main/java/project/cinema/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -31,6 +34,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User userFromDB = userService.add(user);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
        User userFromDb = userService.add(user);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**